core/process/queries/QueryManagerMysqlMAD.php

```diff
@@ -0,0 +1,854 @@
+				CONVERT_TZ(disappear_time, '+00:00', '".self::$time_offset."') AS disappear_time_real,
+				latitude, longitude, cp, individual_attack, individual_defense, individual_stamina
+				FROM pokemon
+				ORDER BY last_modified DESC
+				LIMIT 0,12";
+        $result = $this->mysqli->query($req);
+        $data = array();
+        if ($result->num_rows > 0) {
+            while ($row = $result->fetch_object()) {
+                $data[] = $row;
+            }
+        }
+
+        return $data;
+    }
+
+    public function getRecentMythic($mythic_pokemons)
+    {
+        $req = "SELECT DISTINCT pokemon_id, encounter_id, disappear_time, last_modified,
+				CONVERT_TZ(disappear_time, '+00:00', '".self::$time_offset."') AS disappear_time_real,
+				latitude, longitude, cp, individual_attack, individual_defense, individual_stamina
+				FROM pokemon
+				WHERE pokemon_id IN (".implode(',', $mythic_pokemons).')
+				ORDER BY last_modified DESC
+				LIMIT 0,12';
+        $result = $this->mysqli->query($req);
+        $data = array();
+        if ($result->num_rows > 0) {
+            while ($row = $result->fetch_object()) {
+                $data[] = $row;
+            }
+        }
+
+        return $data;
+    }
+
+    ///////////////////
+    // Single Pokemon
+    ///////////////////
+
+    public function getGymsProtectedByPokemon($pokemon_id)
+    {
+        $req = "SELECT COUNT(DISTINCT(gym_id)) AS total FROM gym WHERE guard_pokemon_id = '".$pokemon_id."'";
+        $result = $this->mysqli->query($req);
+        $data = $result->fetch_object();
+
+        return $data;
+    }
+
+    public function getPokemonLastSeen($pokemon_id)
+    {
+        $req = "SELECT disappear_time,
+				CONVERT_TZ(disappear_time, '+00:00', '".self::$time_offset."') AS disappear_time_real,
+				latitude, longitude
+				FROM pokemon
+				WHERE pokemon_id = '".$pokemon_id."'
+				ORDER BY disappear_time DESC
+				LIMIT 0,1";
+        $result = $this->mysqli->query($req);
+        $data = $result->fetch_object();
+
+        return $data;
+    }
+
+    public function getTop50Pokemon($pokemon_id, $top_order_by, $top_direction)
+    {
+        $req = "SELECT CONVERT_TZ(disappear_time, '+00:00', '".self::$time_offset."') AS distime,
+				pokemon_id, disappear_time, latitude, longitude,
+				cp, individual_attack, individual_defense, individual_stamina,
+				ROUND(100*(individual_attack+individual_defense+individual_stamina)/45,1) AS IV,
+				move_1, move_2, form
+				FROM pokemon
+				WHERE pokemon_id = '".$pokemon_id."' AND move_1 IS NOT NULL AND move_1 <> '0'
+				ORDER BY $top_order_by $top_direction, disappear_time DESC
+				LIMIT 0,50";
+        $result = $this->mysqli->query($req);
+        $top = array();
+        while ($data = $result->fetch_object()) {
+            $top[] = $data;
+        }
+
+        return $top;
+    }
+
+    public function getTop50Trainers($pokemon_id, $best_order_by, $best_direction)
```

Comment:
I don't think we neeed this stuff for  MAD since Mad doesn't store trainerdata, all files that call this function should be edited as well. Same goes for functions: trainer_blacklist, gettrainers, getTrainerLevelCount, getTrainerData,  getTrainerLevelRating, getTrainerActivePokemon, getTrainerInactivePokemon.

Reply:
Ive change it to return null without doing a db lookup, can you test and see how it goes